### ISSUE TITLE
STAR WARS™ Jedi Knight: Dark Forces II (32380)

### ISSUE THREAD

**Issue by kforney** [ValveSoftware/Proton#842]
[System Information](https://gist.github.com/kforney/d58aecb8e8fedfd15e0743643e264eeb)

The game launches, but the menus and cutscene FMVs are in a ~640x480 box in the top-left of my screen.

Menu navigation is difficult, as the mouse cursor doesn't line up with the actual menu buttons.

Finally, when actually in a level, the texture colors are all rainbow, and the resolution is messed up in a different way (the height is correct, but the game runs at about half of my screen's actual resolution) while the HUD displays properly in the corners of the screen.

**Comment by ghost** [comment]
If I recall correctly, the FMVs and menus being stuck in a corner of the screen was also an issue under Windows before, so it may have nothing to do with Proton.

Aside from the resolution issue, the game played fine on [my system](https://gist.github.com/DrWindu/1a10ea5e4305d2f4716bc75fdf6fdfff) with the following workarounds :
- The in-game screen was initially stretch and garbled color-wise. Going to the graphics options menu, enabling 3D acceleration and setting the resolution to the native one of my main screen (1920x1080) fixed the issue instantly.
- I encountered the [same key-binding issue](https://github.com/ValveSoftware/Proton/issues/509#issuecomment-416004993) I had with Dark Forces, making the Bryar un-selectable on my AZERTY layout. However, Jedi Knight lets the player re-bind keys, so hitting my `é`key helpfully bound "Unknown key" to the Bryar pistol, and the key subsequently worked fine in-game for weapon selection.

Testing the game itself, I was able to quickly complete Nar Shadaa(*) with no glaring issues, aside from the menu/cutscene issues described by OP.

(*) Thanks to the supernatural movement speed of 90s shooters. Bless them.

**Comment by kforney** [comment]
Yup, that works.  The menus appear to be stuck in the corner of the screen, but the actual button placement is where you would expect it to be if the menu actually filled the screen properly.

Turning on 3D acceleration fixed the garbled colors/in-game resolution issue immediately.

**Comment by martynhare** [comment]
This game is indeed broken on modern Windows 10.  But *not* on Linux under Steam Play!  Here's a proper workaround for the screen resolution issue which I am hoping a Valve engineer will be able to script as a winetricks-style registry import as a proton-specific workaround:

- Open a terminal
- Load winecfg for the proton prefix, for example:  
  WINEARCH=win64 WINEPREFIX=$HOME/.local/share/Steam/steamapps/compatdata/32380/pfx winecfg
- Go to Graphics tab in winecfg
- Configure it as seen here: ![screenshot from 2018-08-29 20-10-11](https://user-images.githubusercontent.com/5699423/44809822-5b14e780-abbf-11e8-8131-a34ea3563441.png)
- Launch game after as usual
- Enable 3D acceleration and pick a resolution which is just below your native fullscreen resolution
- Enjoy as game window resizes for cutscenes etc. correctly

The "true" workaround is to modify the game files or use a wrapper like dgVoodoo as mentioned here: https://steamcommunity.com/app/32380/discussions/0/490125103626477734/

Technically, Valve staff could legitimately close this as NOTABUG since the game is equally broken on modern Windows as it is under Wine, however, I kinda hope that Valve start making games that don't work on Windows start working on Linux seamlessly instead.  It would be an excellent selling point if Valve could properly GNU Old Games (unlike a certain competitor).

-- Edit: To clarify "Wine" as "Steam Play" (for the uninitiated)

**Comment by waspennator** [comment]
Testing this out on my le oled deck. After getting past the unskippable intro, my deck visually freezes after 5-30 seconds when I try creating a new user profile and i have to manually force the console off. I've been able to get it to occur on experimental and 8.0, but not 7.0.

Saw another report also saying that the game crashes after a few seconds on the main menu, but I haven't gotten that far yet.

[steam-32380.log.txt](https://github.com/ValveSoftware/Proton/files/13809108/steam-32380.log.txt)
